### PR TITLE
OpenVPN: Configurable keepalives, server config comments.

### PR DESCRIPTION
**Pull request by cpu**
I spent some time in the `openvpn` man page today so here are some small Streisand OpenVPN improvements this commit brings:

* Makes the `keepalive` configurable based on a playbook var
* Lowers the default keepalive interval and timeouts: Previously the OpenVPN connection would have to be broken for about an hour before it would be automatically restarted from the client side (and about two hours on the server side!). This commit applies the OpenVPN recommended values and the  connection should be restarted client-side after ~1m. It should be easier to lower/raise this in the future based on battery life/connectivity trade-off if required.
* Adds `mute 3` to the server config: why not :-)
* Adds `mlock` to the server config: this should be a very small security improvement.
* Adds comments for all of the OpenVPN server configuration. I wanted to understand what each directive did, so I figured it out.

### CODE REVIEW

File: playbooks/roles/openvpn/templates/etc_openvpn_server_common.j2 [hunk 1]
```diff
@@ -1,32 +1,87 @@
...
+# It is equivalent to separately setting `ping` and `ping-restart`.
+# On the server the timeout will automatically be adjusted to be 2x as large as
+# the timeout pushed to the client so that the client detects disconnects first.
+keepalive {{ openvpn_keepalive_interval }} {{ openvpn_keepalive_timeout }}
+
+# Encrypt and authenticate all control channel packets with `ta.key`. This is
+# the modern replacement to `tls-auth`. It does not require `key-direction`.
+tls-crypt ta.key
+# Data channel encryption cipher. For OpenVPN clients >2.4 an upgrade to the
+# ncp-cipher understood by both sides.
```
Comment (cpu) [inline playbooks/roles/openvpn/templates/etc_openvpn_server_common.j2]: note to self: I didn't finish this sentence :blush:

**Comment by nopdotcom** [comment]
Looks like the version of Tunnelblick we're mirroring doesn't like this config.

```
Could not start OpenVPN (openvpnstart returned with status #251)

Contents of the openvpnstart log:
*Tunnelblick: openvpnstart log:
     OpenVPN returned with status 1, errno = 0:
          Undefined error: 0
     
     Command used to start OpenVPN (one argument per displayed line):
     
          /Applications/Tunnelblick.app/Contents/Resources/openvpn/openvpn-2.3.18-libressl-2.5.0/openvpn
          --daemon
          --log
          /Library/Application Support/Tunnelblick/Logs/-SUsers-Suser-SLibrary-SApplication Support-STunnelblick-SConfigurations-S192.0.2.1--combined.tblk-SContents-SResources-Sconfig.ovpn.769_0_1_0_1065264.1337.openvpn.log
          --cd
          /Library/Application Support/Tunnelblick/Users/user/192.0.2.1-combined.tblk/Contents/Resources
          --verb
          3
          --config
          /Library/Application Support/Tunnelblick/Users/user/192.0.2.1-combined.tblk/Contents/Resources/config.ovpn
          --verb
          3
          --cd
          /Library/Application Support/Tunnelblick/Users/user/192.0.2.1-combined.tblk/Contents/Resources
          --management
          127.0.0.1
          1337
          --management-query-passwords
          --management-hold
          --script-security
          2
          --up
          /Applications/Tunnelblick.app/Contents/Resources/client.up.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw
          --down
          /Applications/Tunnelblick.app/Contents/Resources/client.down.tunnelblick.sh -9 -d -f -m -w -ptADGNWradsgnw
     
     Contents of the OpenVPN log:
     
          Mon Mar 26 19:55:55 2018 Option 'nobind' in /Library/Application Support/Tunnelblick/Users/user/192.0.2.1-combined.tblk/Contents/Resources/config.ovpn:11 is ignored by previous <connection> blocks 
          Options error: Unrecognized option or missing parameter(s) in /Library/Application Support/Tunnelblick/Users/user/192.0.2.1-combined.tblk/Contents/Resources/config.ovpn:27: tls-crypt (2.3.18)
          Use --help for more information.
          
     More details may be in the Console Log's "All Messages"
```

**Comment by nopdotcom** [comment]
@alimakki might have some insight on this.

**Comment by cpu** [comment]
> Looks like the version of Tunnelblick we're mirroring doesn't like this config.

It looks like it doesn't understand `tls-crypt` and was broken by https://github.com/StreisandEffect/streisand/commit/8ba56be8e912d59fe0b7a8cad73d1d72d65fd3df It won't work with master either :-(

**Comment by nopdotcom** [comment]
I'll file an issue.

**Comment by alimakki** [comment]
I think Tunnelblick bundles 2 version of OpenVPN, 2.3 and 2.4. It defaults to 2.3 if I recall correctly.

**Comment by alimakki** [comment]
screen shot 2018-03-26 at 8 42 21 pm

**Comment by nopdotcom** [comment]
It bundles *a lot* of versions:

capto_capture 2018-03-26_08-43-09_pm

**Comment by cpu** [comment]
@alimakki @nopdotcom Is there something in the Tunnelblick docs I should change to point folks to using a 2.4.4 based OpenVPN option with either LibreSSL or OpenSSL?

**Comment by alimakki** [comment]
@cpu created #1239 to track this.

**Comment by cpu** [comment]
Thanks @alimakki 

I'm going to merge this PR with @nopdotcom's +1. The issue flagged in #1239 is unfortunate but since its in master already I don't think we should hold this PR up on a fix.

**Pull request merged**